java/vector/src/main/java/org/apache/arrow/vector/BaseVariableWidthViewVector.java

```diff
@@ -0,0 +1,1434 @@
+      setBytes(i, EMPTY_BYTE_ARRAY, 0, EMPTY_BYTE_ARRAY.length);
+    }
+    lastSet = index - 1;
+  }
+
+  /**
+   * Get the length of the view.
+   * @param index The index of the element in the vector.
+   * @return The length of the element at the given index.
+   */
+  protected final int getLength(int index) {
+    if (index < 0 || index >= viewBuffer.capacity() / ELEMENT_SIZE) {
+      throw new IndexOutOfBoundsException("Index out of bounds: " + index);
+    }
+    if (isSet(index) == 0) {
+      return 0;
+    }
+    return viewBuffer.getInt(((long) index * ELEMENT_SIZE));
+  }
+
+  protected ArrowBuf allocateOrGetLastBuffer(BufferAllocator allocator, int length, List<ArrowBuf> dataBuffers) {
+    long dataBufferSize;
+    if (initialDataBufferSize > 0) {
+      dataBufferSize = initialDataBufferSize;
+    } else {
+      dataBufferSize = lastValueAllocationSizeInBytes;
+    }
+
+    if (dataBuffers.isEmpty() || dataBuffers.get(dataBuffers.size() - 1).capacity() -
+            dataBuffers.get(dataBuffers.size() - 1).writerIndex() < length) {
+      ArrowBuf newBuf = allocator.buffer(dataBufferSize);
+      dataBuffers.add(newBuf);
+    }
+
+    return dataBuffers.get(dataBuffers.size() - 1);
+  }
+
+  /**
+   * This method is used to create a view buffer for a variable width vector.
+   * It handles both inline and data buffers.
+   * <p>
+   * If the length of the value is less than or equal to {@link #INLINE_SIZE}, the value is stored in the valueBuffer
+   * directly as an inline buffer.
+   * The valueBuffer stores the length of the value followed by the value itself.
+   * If the length of the value is greater than {@link #INLINE_SIZE}, a new buffer is allocated and added to dataBuffers
+   * to hold the value.
+   * The viewBuffer in this case stores the length of the value, a prefix of the value, the index of the
+   * new buffer in dataBuffers, and the offset of the value in the new buffer.
+   *
+   * @param allocator The allocator used to create new ArrowBuf instances.
+   * @param index The index at which the new value will be inserted.
+   * @param value The byte array that contains the data to be inserted.
+   * @param start The start index in the byte array from where the data for the new value begins.
+   * @param length The length of the data in the byte array that belongs to the new value.
+   * @param viewBuffer The ArrowBuf instance that will hold the data of the new value.
+   * @param dataBuffers The list of ArrowBuf instances that hold the data of all long binary values in the vector.
+   */
+  protected void createViewBuffer(
+          BufferAllocator allocator,
+          int index,
+          byte[] value,
+          int start,
+          int length,
+          ArrowBuf viewBuffer,
+          List<ArrowBuf> dataBuffers) {
+    int writePosition = index * ELEMENT_SIZE;
+    if (value.length <= INLINE_SIZE) {
+      // allocate inline buffer
+      // set length
+      viewBuffer.setInt(writePosition, length);
```

Comment:
this can be done outside the if/else?

Reply:
We can leave setting the length as a common part, but I would rather like to leave it as it is for readability in a sense that, for a short string this is how we allocate vs allocation for a long string. But if it needs moving, we can do, not a problem :slightly_smiling_face: